### DISCUSSION TITLE
supabase flutter doesn't throw error on duplicate account registration

### DISCUSSION THREAD

**Discussion by dukesx** [supabase/supabase#4005] [Questions] [answered]
I am doing a sign up with a phone number and password. 

Usually it is expected that if account already exists, client should throw error of duplication but it seems that supabase does not throw error at all. Is this normal with supabase flutter? 

Anyone help?

**Comment by dukesx** [comment]
Can someone please help ?

**Comment by awalias** [comment, marked as answer]
hey @dukesx ! Some discussion on this issue here: https://github.com/supabase/supabase-js/issues/296

tl;dr it's not secure to give this info on a public endpoint - solutions/workarounds are:
- disable email confirmations (not advised)
- wait for us to implement [a better email flow](https://github.com/supabase/gotrue/issues/286) for duplicate signups (or manually trigger a password reset email)
- sign the user up using the admin endpoint (server side only https://github.com/supabase/supabase-js/issues/296#issuecomment-976203549)
- make a separate call to .user() to find out if the account exists (and it confirmed), also server side only

**Reply by dukesx** [reply]
But what I do not understand is, I am using phone number instead of email. Those discussions are on email verification while mine is on phone number. Anyhow its I will try with user() to check for duplicate
